components/payments/settings/payment-webhooks.tsx

```diff
@@ -7,5 +21,210 @@ interface PaymentsWebhooksProps {
 export const PaymentsWebhooks: React.FC<PaymentsWebhooksProps> = ({
   accountId,
 }) => {
-  return <Flex flexDir="column" gap={8}></Flex>;
+  const { data: webhookApiKey, isLoading: isLoadingSecretKey } =
+    usePaymentsWebhooksSecretKeyByAccountId(accountId);
+  const {
+    data: webhooks,
+    isLoading: isGetLoading,
+    isFetched,
+  } = usePaymentsWebhooksByAccountId(accountId);
+  const { mutate: updateWebhook, isLoading: isUpdateLoading } =
+    usePaymentsUpdateWebhook(accountId);
+  const { mutate: createWebhook, isLoading: isCreateLoading } =
+    usePaymentsCreateWebhook(accountId);
+
+  const toast = useToast();
+
+  const triggerAlert = (
+    status: "error" | "success",
+    header: string,
+    description: string,
+  ) => {
+    toast({
+      position: "bottom",
+      variant: "solid",
+      title: header,
+      description,
+      status,
+      duration: 9000,
+      isClosable: true,
+    });
+  };
+
+  const { productionWebhooks, testnetWebhooks } = useMemo(() => {
+    if (webhooks) {
+      const productionWebhooks = webhooks.filter(
+        (webhook) => webhook.isProduction,
+      );
+      const testnetWebhooks = webhooks.filter(
+        (webhook) => !webhook.isProduction,
+      );
+
+      return { productionWebhooks, testnetWebhooks };
+    }
+    return { productionWebhooks: [], testnetWebhooks: [] };
+  }, [webhooks]);
+
+  const onUpdateWebhook: PaymentsWebhooksTableProps["onUpdate"] = (
+    webhook,
+    newUrl,
+  ) => {
+    if (!isValidWebhookUrl(newUrl)) {
+      triggerAlert(
+        "error",
+        "Invalid Webhook Url",
+        `${newUrl} is not a valid webhook url, please try a different url`,
+      );
+      return;
+    }
+
+    // send the request
+    updateWebhook(
+      { webhookId: webhook.id, url: newUrl },
+      {
+        onSuccess: () => {
+          triggerAlert(
+            "success",
+            "Webhook Created",
+            `Successfully created  ${
+              webhook.isProduction ? "production" : "testnet"
+            } webhook: ${webhook.url}`,
+          );
+        },
+        onError: () => {
+          triggerAlert(
```

Comment:
add the onError from useTxNotifications and tracking:

`              trackEvent({
                category: "payments",
                action: "delete-webhook",
                label: "error",
                error,
              });`

Reply:
I also feel you could every easily merge delete and update if you wanted but that's something we can look over tomorrow, it's a nice code optimization